doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+        "PPO",
+        param_space=config,
+        run_config=train.RunConfig(
+            stop={
+                metric: 450.0,
+            },
+            name="docs_rllib_offline_pretrain_ppo",
+            verbose=2,
+            checkpoint_config=train.CheckpointConfig(
+                checkpoint_frequency=1,
+                checkpoint_at_end=True,
+            ),
+        ),
+    )
+    analysis = tuner.fit()
 
-    For custom models and enviroments, you'll need to use the `Python API <rllib-training.html#basic-python-api>`__.
+    best_checkpoint = (
+        analysis
+        .get_best_result(
+            metric=metric,
+            mode="max"
+        )
+        .checkpoint.path
+    )
 
-In this example, we will save batches of experiences generated during online training to disk, and then leverage this saved data to train a policy offline using DQN. First, we run a simple policy gradient algorithm for 100k steps with ``"output": "/tmp/cartpole-out"`` to tell RLlib to write simulation outputs to the ``/tmp/cartpole-out`` directory.
+Example: Record expert data to local disk
+-----------------------------------------
+After you trained an expert policy to play ``CartPole-v1`` you load its policy here to record expert data during evaluation. You will use 
+5 ``OfflineSingleAgentEnvRunner`` s to collect in each rollout 50 complete episodes. In this example you store experiences directly in RLlib's 
+``SingleAgentEpisode`` objects with no more than 25 episode objects per Parquet file. Altogether you run 10 evaluation runs, which should
+result in 500 recorded episodes from the expert policy. You will use these data in the next example to train via Offline RL a new policy 
+that should reach a return of 450.0 when playing ``CartPole-v1``.
+
+.. testcode::
+
+    from ray.rllib.algorithms.ppo import PPOConfig
+    from ray.rllib.core import (
+        COMPONENT_LEARNER_GROUP,
+        COMPONENT_LEARNER,
+        COMPONENT_RL_MODULE,
+        DEFAULT_MODULE_ID,
+    )
+    from ray.rllib.core.rl_module import RLModuleSpec
+    
+    # Store recording data under the following path.
+    data_path = "/tmp/docs_rllib_offline_recording"
 
-.. code-block:: bash
+    # Configure the algorithm for recording.
+    config = (
+        PPOConfig()
+        # The environment needs to be specified.
+        .environment(
+            env="CartPole-v1",
+        )
+        # Make sure you use the new API stack.
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        # Make sure to sample complete episodes because
+        # you want to record RLlib's episode objects.
+        .env_runners(
+            batch_mode="complete_episodes",
+        )
+        # Set up 5 evaluation `EnvRunners` for recording.
+        # Sample 50 episodes in each evaluation rollout.
+        .evaluation(
+            evaluation_num_env_runners=5,
+            evaluation_duration=50,
```

Comment:
let's also specify the unit here, just for clarity `evaluation_duration_unit="episodes"`

Reply:
Alright, implemented it! Thanks for the suggestion!